From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Akanksha Hotta <akanksha.hotta@intel.com>
Date: Wed, 10 Jun 2026 14:13:37 +0530
Subject: drm: Add drm_gem_gpuva struct compatibility for lockdep
 config

Keep the imported GEM object's gpuva field compatible with target kernels.
Only the shared GEM header is patched; no GPUVM implementation is imported.

Reference:
   e7fa80e2932c
   drm_gem: add mutex to drm_gem_object.gpuva
   3c8d31b8937a
   gpuvm: remove gem.gpuva.lock_dep_map

Signed-off-by: Akanksha Hotta <akanksha.hotta@intel.com>
---

diff --git a/include/drm/drm_gem.h b/include/drm/drm_gem.h
--- a/include/drm/drm_gem.h
+++ b/include/drm/drm_gem.h
@@ -428,7 +428,16 @@
 		 * do not allocate memory while holding this lock. Otherwise,
 		 * the &dma_resv lock should be used.
 		 */
+#ifdef BPM_DRM_GEM_GPUVA_MUTEX_NOT_PRESENT
+#ifdef CONFIG_LOCKDEP
+#ifndef BPM_DRM_GEM_GPUVA_LOCK_DEP_MAP_NOT_PRESENT
+		struct lockdep_map *lock_dep_map;
+#endif
+#endif
+#else
 		struct mutex lock;
+#endif
+
 	} gpuva;

 	/**
@@ -649,10 +662,23 @@
 }

 #ifdef CONFIG_LOCKDEP
+#ifdef BPM_DRM_GEM_GPUVA_MUTEX_NOT_PRESENT
+#ifdef BPM_DRM_GEM_GPUVA_LOCK_DEP_MAP_NOT_PRESENT
+#define drm_gem_gpuva_assert_lock_held(gpuvm, obj) \
+	lockdep_assert(drm_gpuvm_immediate_mode(gpuvm) ? 1 : \
+		       dma_resv_held((obj)->resv))
+#else
+#define drm_gem_gpuva_assert_lock_held(gpuvm, obj) \
+	lockdep_assert(drm_gpuvm_immediate_mode(gpuvm) ? \
+		       ((obj)->gpuva.lock_dep_map ? lock_is_held((obj)->gpuva.lock_dep_map) : 1) : \
+		       dma_resv_held((obj)->resv))
+#endif
+#else
 #define drm_gem_gpuva_assert_lock_held(gpuvm, obj) \
 	lockdep_assert(drm_gpuvm_immediate_mode(gpuvm) ? \
 		       lockdep_is_held(&(obj)->gpuva.lock) : \
 		       dma_resv_held((obj)->resv))
+#endif
 #else
 #define drm_gem_gpuva_assert_lock_held(gpuvm, obj) do {} while (0)
 #endif

-- 
2.43.0

